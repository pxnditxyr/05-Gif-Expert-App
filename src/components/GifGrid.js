// import React, { useState, useEffect } from 'react';
import React from 'react';
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
// import { GifGridItem } from './GifGridItem';
// import { getGifs } from '../helpers/getGifs';


export const GifGrid = ({ category }) => {


	const { data, loading } = useFetchGifs();
	console.log( data );
	console.log( loading );
	// const [ images, setImages ] = useState( [] );
	// useEffect( () => {
	// 	getGifs( category )
	// 		.then( imgs => {
	// 			setImages( imgs );
	// 		});
	// }, [ category ])


	return (
		<>
			<h2> { category } </h2>
			{ loading ? 'Loading...' : 'Data charge done' }
			{/* <div className="card-grid"> */}
			{/* 	{ */}
			{/* 		images.map( img => ( */}
			{/* 			<GifGridItem */}
			{/* 				key={ img.id } */}
			{/* 				{ ...img  } */}
			{/* 			/> */}
			{/* 		)) */}
			{/* 	} */}
			{/* </div> */}
		</>
	);
};

GifGrid.prototype = {
	category: PropTypes.string.isRequired,
}
GifGrid.defaultProps = {
	category: "Nuevo"
}
