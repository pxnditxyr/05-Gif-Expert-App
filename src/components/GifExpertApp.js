import React, { useState } from 'react';
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';

export const GifExpertApp = () => {

	// const categories = [ "Naruto", "Bleach", "One Piece" ];
	//const [ categories, setCategories ] = useState( ["Naruto", "Bleach", "One Piece", "KissxSis"] );
	const [ categories, setCategories ] = useState( ["Naruto"] );


	// const handleAdd = () => setCategories( [...categories, "KissxSis"] )

	return (

		<>
			<h2> GifExpertApp XD </h2>
			<AddCategory setCategories={ setCategories } />
			<hr />
			{/* <button onClick={ handleAdd }> Add </button> */}
			<ol>
				{
					categories.map( category =>
						<GifGrid
							key={ category }
							category={ category }
						/>
						//{/* <li key={ category }> { category } </li> */}
					)
				}
			</ol>
		</>
	);
};
