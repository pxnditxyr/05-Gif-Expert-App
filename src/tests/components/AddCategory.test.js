import { AddCategory } from '../../components/AddCategory';
import { shallow } from 'enzyme';

describe( "Tests on AddCategory component", () => {

	const setCategories = jest.fn();
	let wrapper;

	beforeEach( () => {
		jest.clearAllMocks()
		wrapper = shallow( <AddCategory setCategories={ setCategories }/> );
	});

	test( "Should return ", () => {
		expect( wrapper ).toMatchSnapshot();
	});

	test( "Should change the text box", () => {
		const input = wrapper.find( 'input' );
		const value = "Hello my love!!"

		input.simulate( 'change', { target: { value } });
		const p = wrapper.find( 'p' );
		expect( p.text().trim() ).toBe( value );
	});


	test( "Dont should submit changes", () => {
		const form = wrapper.find( 'form' );
		form.simulate( 'submit', { preventDefault(){} } );
		expect( setCategories ).not.toHaveBeenCalled();
	});

	test( "Should call setCategories and clear text box", () => {

		const value = "Hello my love!!"

		wrapper.find( 'input' ).simulate( 'change', { target: { value } });

		wrapper.find( 'form' ).simulate( 'submit', { preventDefault(){} } );

		expect( setCategories ).toHaveBeenCalled();
		expect( setCategories ).toHaveBeenCalledTimes( 1 );
		expect( setCategories ).toHaveBeenCalledWith( expect.any( Function ) );

		expect( wrapper.find( 'input' ).prop( 'value' ) ).toBe( "" );


	});
});

