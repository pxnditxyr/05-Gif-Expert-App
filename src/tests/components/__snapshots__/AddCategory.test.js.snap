// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on AddCategory component Should return  1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
